frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { Link } from 'react-router-dom';
+import { Tooltip, Button } from '@patternfly/react-core';
+import { TopologyPage, renderTopology } from '../TopologyPage';
+import NamespacedPage from '../../NamespacedPage';
+import { AsyncComponent, StatusBox } from '@console/internal/components/utils';
+import ConnectedTopologyDataController from '../TopologyDataController';
+import ProjectListPage from '../../projects/ProjectListPage';
+import { topologyData } from './topology-test-data';
+import Topology from '../Topology';
+
+type TopologyPageProps = React.ComponentProps<typeof TopologyPage>;
+type RenderTopologyProps = React.ComponentProps<typeof renderTopology>;
+
+let topologyProps: TopologyPageProps;
+let renderTopologyProps: RenderTopologyProps;
+
+describe('Topology page tests', () => {
+  beforeEach(() => {
+    topologyProps = {
+      activeApplication: 'topology-test',
+      knative: false,
+      cheURL: 'che.openshift.io',
+      serviceBinding: false,
+      match: {
+        params: {
+          name: 'topology-test',
+        },
+        isExact: true,
+        path: '',
+        url: '',
+      },
+    };
+
+    renderTopologyProps = {
+      data: topologyData,
+      loaded: true,
+      loadError: '',
+      serviceBinding: false,
+    };
+  });
+
+  it('should render topology list page', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(AsyncComponent).exists()).toBe(true);
+  });
+
+  it('should render topology graph page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ConnectedTopologyDataController).exists()).toBe(true);
+  });
+
+  it('should render projects list page', () => {
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ProjectListPage).exists()).toBe(true);
+  });
+
+  it('should render view shortcuts button on topology page toolbar', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).props().children).toEqual('View shortcuts');
+  });
+
+  it('should not render view shortcuts button on topology list page toolbar', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).exists()).toBe(false);
```

Comment:
I see what you're trying to do with this test and the previous test... but blindly looking for a `Button` is a bad way to go about it. How do you know this button is the right button and `NamespacedPage` isn't just adding buttons for their own use-cases?

Perhaps find a better way to identify this button... perhaps put a `data-test-id` on it?

@christianvogt and I have been mulling the idea around testing for `className`s for validation of using css - more-so as a poor-mans "make sure you clean up css after you stop using the class". But I don't know if that's the best approach here to identify the button.

Reply:
Add a `data-test-id`

Reply:
added ```data-test-id```.